src/blocks/nordvpn.rs

```diff
@@ -0,0 +1,219 @@
+//! Current connection status for nordvpn networks
+//!
+//! Behind the scenes this uses the `nordvpn` command line binary. In order for this to work
+//! properly the binary should be executable without root privileges.
+//!
+//! # Configuration
+//!
+//! Key | Values | Default
+//! ----|--------|--------
+//! `interval` | Update interval in seconds. | `10`
+//! `format_connected` | A string to customise the output in case the network is connected. See below for available placeholders. | `" VPN: $icon "`
+//! `format_disconnected` | A string to customise the output in case the network is disconnected. See below for available placeholders. | `" VPN: $icon "`
+//! `state_connected` | The widgets state if the nordvpn network is connected. | `info`
+//! `state_disconnected` | The widgets state if the nordvpn network is disconnected | `idle`
+//!
+//! Placeholder | Value                          | Type   | Unit
+//! ------------|--------------------------------|--------|------
+//! `icon`      | A static icon                  | Icon   | -
+//! `country`   | Country currently connected to | Text   | -
+//!
+//! # Example
+//!
+//! Shows the current nordvpn network state:
+//!
+//! ```toml
+//! [[block]]
+//! block = "nordvpn"
+//! interval = 10
+//! format_connected = "VPN: $icon "
+//! format_disconnected = "VPN: $icon "
+//! state_connected = "info"
+//! state_diconnected = "idle"
+//! ```
+//!
+//! Possible values for `state_connected` and `state_diconnected`:
+//!
+//! ```
+//! warning
+//! critical
+//! info
+//! idle
+//! ```
+//! `Option::None` defaults to `idle`.
+//!
+//! # Icons Used
+//!
+//! - `net_vpn`
+//! - `net_wired`
+//! - `net_down`
+
+use core::str::FromStr;
+use std::process::Stdio;
+use tokio::process::Command;
+
+use super::prelude::*;
+
+#[derive(Deserialize, Debug)]
+#[serde(default)]
+pub struct Config {
+    interval: Seconds,
+    format_connected: FormatConfig,
+    format_disconnected: FormatConfig,
+    state_connected: Option<String>,
+    state_disconnected: Option<String>,
+}
+
+impl Default for Config {
+    fn default() -> Config {
+        Config {
+            interval: 10.into(),
+            format_connected: FormatConfig::from_str(" VPN: $icon ").unwrap(),
+            format_disconnected: FormatConfig::from_str(" VPN: $icon ").unwrap(),
+            state_connected: Option::from(String::from("info")),
+            state_disconnected: Option::from(String::from("idle")),
+        }
+    }
+}
+
+enum Status {
+    Connected(String),
+    Disconnected,
+    Error,
+}
+
+impl Status {
+    fn icon(&self) -> &str {
+        match *self {
+            Status::Connected(_) => "net_vpn",
+            Status::Disconnected => "net_wired",
+            Status::Error => "net_down",
+        }
+    }
+}
+
+pub async fn run(config: Config, mut api: CommonApi) -> Result<()> {
+    api.set_default_actions(&[(MouseButton::Left, None, "toggle")])
```

Comment:
Please document "toggle" action.

Reply:
Would you consider [this](https://github.com/greshake/i3status-rust/pull/1725/commits/2e502dcd794ea28a7c16ebce0badc26e3ad5d0ee#diff-3f790987dd77e1f06baa6dcaf001d026de91ede112d783a6eb2e37b73fbc1885R2) a documentation?
If not, please suggest how you would document this.

Reply:
Otherwise consider [this](https://github.com/greshake/i3status-rust/pull/1725/commits/4670d15801dc6caba8162d06e1b8c1446f62d2d0#diff-3f790987dd77e1f06baa6dcaf001d026de91ede112d783a6eb2e37b73fbc1885R21-R24)

Reply:
> Otherwise consider [this](https://github.com/greshake/i3status-rust/pull/1725/commits/4670d15801dc6caba8162d06e1b8c1446f62d2d0#diff-3f790987dd77e1f06baa6dcaf001d026de91ede112d783a6eb2e37b73fbc1885R21-R24)

Yes, exposing the name of the action lets users remap it to different buttons.